prov/rxm/src/rxm.h

```diff
@@ -420,12 +420,13 @@ enum rxm_buf_pool_type {
 };
 
 struct rxm_buf {
-	/* Must stay at top */
-	struct fi_context fi_context;
-
-	enum rxm_proto_state state;
-
-	void *desc;
+	/* Must stay at top. */
+	struct fi_context	fi_context; /* support FI_CONTEXT mode */
+#if ENABLE_DEBUG
+	uint64_t 		magic;
+#endif
```

Comment:
Would it help to just put something like this directly into the buffer pool abstraction?

Reply:
Hmm. That would require accessing fields of the buffer pool abstraction as we need to do the assert check when we get a completion. I think users of buffer pool don't access those fields to avoid dependency on the implementation.

Btw, it seems the assert's getting executed and failing in windows release builds. Any ideas to resolve that?

Reply:
I was thinking of setting/clearing the value as the buffer is removed/returned to the pool.  Maybe this can get pushed down into the list abstractions?

For this change, if you want to track from posting to completion, I would clear the magic once the buffer completes.

Along this line, maybe create some helper macros, so we don't need to sprinkle the if ENABLE_DEBUG through the code.  E.g. (rough outline)
```
#if ENABLE_DEBUG
    #define OFI_DECLARE_MAGIC uint64_t magic
    #define OFI_SET_MAGIC(ptr) ptr->magic = OFI_MAGIC_64
    #define OFI_CLEAR_MAGIC(ptr) ptr->magic =~OFI_MAGIC_64
    #define OFI_VERIFY_MAGIC(ptr) assert(ptr->magic == OFI_MAGIC_64)
#else
    #define OFI_DECLARE_MAGIC
    #define OFI_SET_MAGIC(ptr)
    #define OFI_CLEAR_MAGIC(ptr)
    #define OFI_VERIFY_MAGIC(ptr)
#endif
```

Reply:
I should perhaps restrict this magic check to the debug hooking provider. I can add these macros there.